File: 0124-binary-tree-maximum-path-sum/0124-binary-tree-maximum-path-sum.java
class Solution {

    int ans = Integer.MIN_VALUE;
    private int find(TreeNode root) {
        if(root == null) 
        return 0;

        int l = Math.max(0,find(root.left));
        int r = Math.max(0,find(root.right));

        int curr = l + r + root.val;
        ans = Math.max(ans,curr);

        return Math.max(l,r) + root.val;
    }
    public int maxPathSum(TreeNode root) {
        find(root);
        return ans;
    }
}

File: 0124-binary-tree-maximum-path-sum/0124-binary-tree-maximum-path-sum.cpp
class Solution {
    int ans = INT_MIN;
    
    private:
    int find(TreeNode* root) {
        if(root == nullptr) return 0;
        
        int l = std::max(0, find(root->left));
        int r = std::max(0, find(root->right));
        
        int curr = l + r + root->val;
        ans = std::max(ans, curr);
        
        return std::max(l, r) + root->val;
    }
    
public:
    int maxPathSum(TreeNode* root) {
        ans = INT_MIN; // Reset for multiple test cases if needed
        find(root);
        return ans;
    }
};